nixos/lib/make-bootloader-options.nix

```diff
@@ -0,0 +1,35 @@
+{ lib }:
+/* Build a bootloader set of options */
```

Comment:
I'd prefer a module instead of module generation. Morally it definitely should, but not sure if the existing option structure and its use are receptive to a change to submodules. (Not in `attrsOf`; just submodules)

Reply:
"it definitely should" → should what?
I don't really understand the rest of the sentence of the existing option structure and changes to submodules, I would appreciate you clarifying what you meant.

Reply:
> "it definitely should" →

be a module, morally

"Morally" does a lot of heavy lifting in that sentence, which I could be a bit confusing I guess.

Basically what I'm proposing is to change e.g. `boot.loader.systemd-boot` etc to be a submodule, each of which can individually import a module with just these options at the top level.

It does change the structure of the `options` module argument a bit, changing e.g. `options.boot.loader.systemd-boot` to an option instead of a tree of options. Not sure what's the trade-off here; whether that's a change worth making.

Let's consider this idea not a blocker.

Reply:
Logged in the referred issue. Resolving in the meantime.